tests/std/tests/P0009R18_mdspan_default_accessor/test.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <mdspan>
+#include <string>
+#include <type_traits>
+
+using namespace std;
+
+template <class ElementType>
+constexpr void test_one(array<ElementType, 3> elems) {
+    using DefaultAccessor = default_accessor<ElementType>;
+
+    // Check modeled concepts
+    static_assert(is_trivially_copyable_v<DefaultAccessor>);
+    static_assert(semiregular<DefaultAccessor>);
+
+    // Check nested types
+    static_assert(same_as<typename DefaultAccessor::offset_policy, DefaultAccessor>);
+    static_assert(same_as<typename DefaultAccessor::element_type, ElementType>);
+    static_assert(same_as<typename DefaultAccessor::reference, ElementType&>);
+    static_assert(same_as<typename DefaultAccessor::data_handle_type, ElementType*>);
+
+    // Check default constructor
+    DefaultAccessor accessor;
+    static_assert(is_nothrow_default_constructible_v<DefaultAccessor>);
+
+    { // Check converting constructor from other accessor
+        [[maybe_unused]] default_accessor<const ElementType> const_accessor = accessor;
+        static_assert(is_nothrow_constructible_v<default_accessor<const ElementType>, DefaultAccessor>);
+        static_assert(!is_constructible_v<DefaultAccessor, default_accessor<const ElementType>>);
+    }
+
+    { // Check 'access' member function
+        same_as<ElementType&> decltype(auto) ref = accessor.access(elems.data(), 1);

```

Comment:
This potentially shadows `<functional>`'s `std::ref`. I recommend choosing any other name.